addons/web/static/src/views/fields/relational_utils.js

```diff
@@ -820,5 +845,27 @@ export function useX2ManyCrud(getList, isMany2Many) {
         saveRecord,
         updateRecord,
         removeRecord,
+        openFullRecord
     };
 }
+
+function getDefaultValuesFromRecord(data, fields) {
```

Comment:
Hello @awa-odoo ,
Thanks for your review.
Yes, this method is necessary. It's not used for creating new records. It will set the context value according to field type and create a context dictionary with 'default_[field_name]' key so we can pass them into form view.